include/uart_blob_tx.h

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef UART_BLOB_TX_H__
+#define UART_BLOB_TX_H__
+
+#include <zephyr.h>
+
+
+/**
+ * @file uart_blob_tx.h
+ * @defgroup uart_blob_tx UART BLOB sender.
+ * @{
+ * @brief UART BLOB sender API.
+ */
+
+
+/*****************************************************************************
+ * Structure definitions
+ *****************************************************************************/
+
+/**
+ * @brief UART BLOB sender event type.
+ */
+enum uart_blob_tx_evt_type {
+	/**
+	 * @brief BLOB transfer initiated.
+	 */
+	UART_BLOB_TX_EVT_STARTED = 0,
+
+	/**
+	 * @brief BLOB transfer stopped either due to being finished or
+	 *        due to a protocol error.
+	 */
+	UART_BLOB_TX_EVT_STOPPED,
+
+	/**
+	 * @brief Number of sender events.
+	 */
+	UART_BLOB_TX_EVT_COUNT
+};
+
+/**
+ * @brief UART BLOB sender protocol events.
+ */
+enum uart_blob_tx_err {
+	/**
+	 * @brief BLOB transfer closed by user.
+	 */
+	UART_BLOB_TX_SUCCESS 		= 0,
+
+	/**
+	 * @brief BLOB transfer aborted due to timeout.
+	 */
+	UART_BLOB_TX_ERR_TIMEOUT	= -ETIMEDOUT,
+
+	/**
+	 * @brief BLOB transfer aborted due to UART break error.
+	 */
+	UART_BLOB_TX_ERR_BREAK		= -ENETDOWN,
+
+	/**
+	 * @brief BLOB transfer aborted due to unhandled fatal error.
+	 */
+	UART_BLOB_TX_ERR_FATAL		= -EFAULT
+};
+
+/**
+ * @brief UART BLOB sender event structure.
+ */
+struct uart_blob_tx_evt {
+	/**
+	 * @brief Type of event.
+	 */
+	enum uart_blob_tx_evt_type type;
+
```

Comment:
This look similar to the `RX_` equivalent, would it be possible to extract this to a common shared headerfile?